content/reference/stacks.md

```diff
@@ -0,0 +1,277 @@
+
+# Stack Schema
+
+<!--
+TODO: Is the vault backend supported as of today?
+TODO: Should we mention the `remote` backend considering the current state of the TFC automation module?
+-->
+
+[The official JSON Schema document for Stacks can be found here](https://github.com/cloudposse/atmos/blob/master/docs/schema/stack-config-schema.json). The below is a walk through of a complete example utilizing all capabilities.
+
+```yaml
+# stacks/ue2-dev.yaml
+s3-bucket:
+      vars:
+        enabled: false
+        user_enabled: false
+        acl: "private"
+        grants: null
+        policy: ""
+        force_destroy: false
+        versioning_enabled: true
+        allow_encrypted_uploads_only: false
+        lifecycle_rule_enabled: false
+        noncurrent_version_glacier_transition_days: 30
+        noncurrent_version_deeparchive_transition_days: 60
+        noncurrent_version_expiration_days: 90
+        standard_transition_days: 30
+        glacier_transition_days: 60
+        deeparchive_transition_days: 90
+        enable_glacier_transition: false
+        enable_deeparchive_transition: false
+        enable_standard_ia_transition: false
+        enable_current_object_expiration: false
+        expiration_days: 90
+        lifecycle_tags: {}
+        block_public_acls: true
+        block_public_policy: true
+        ignore_public_acls: true
+        restrict_public_buckets: true
+
+# Imports enables shared configuration / settings across different stacks
+# The referenced files are deep merged into this stack to support granular configuration capabilities
+imports:
+  # Merge the below `stacks/catalog/*.yaml` files into this stack to provide any shared `vars` or `components.*` configuration
+  - catalog/globals
+  - catalog/ue2-globals
+  - catalog/dev-globals
+
+# `vars` provides shared configuration for all components -- both terraform + helmfile
+vars:
+  # Used to determine the name of the workspace (e.g. the 'dev' in 'ue2-dev')
+  stage: dev
+  # Used to determine the name of the workspace (e.g. the 'ue2' in 'ue2-dev')
+  environment: ue2
+
+# Define cross-cutting terraform configuration
+terraform:
+  # `terraform.vars` provides shared configuration for terraform components
+  vars: {}
+
+  # `backend_type` + `backend` provide configuration for the terraform backend you
+  # would like to use for all components. This is typically defined in `globals.yaml`
+  backend_type: s3 # s3, remote, vault
+  backend:
+    s3:
+      encrypt: true
+      bucket: "eg-uw2-root-tfstate"
+      key: "terraform.tfstate"
+      dynamodb_table: "eg-uw2-root-tfstate-lock"
+      role_arn: "arn:aws:iam::999999999999:role/eg-gbl-root-terraform"
+      acl: "bucket-owner-full-control"
+      region: "us-east-2"
+    remote: {}
+    vault: {}
```

Comment:
This is showing how multiple backends can be configured, of which one is enabled with `backend_type`

Reply:
Will mention 👍